test/tests/amqp/test_amqp_racknode.py

```diff
@@ -0,0 +1,205 @@
+        queue_name = result.method.queue
+        self.channel.queue_bind(
+            exchange=exchange_name,
+            queue=queue_name,
+            routing_key=topic_routing_key)
+        self.channel.basic_consume(external_callback, queue=queue_name)
+        self.connection.add_timeout(timeout, self.panic)
+
+    def panic(self):
+        self.channel.stop_consuming()
+        logs.debug_7('Pika connection timeout')
+        self.connection.close()
+        exit(0)
+
+    def run(self):
+        logs.debug_7('start consuming')
+        self.channel.start_consuming()
+
+
+@attr(all=True, regression=True, smoke=True)
+class amqp_rack_node(fit_common.unittest.TestCase):
+    # clear the test environment
+    def tear_down(self):
+        self.test_api_delete_rack()
+
+    def _compare_message(self, expected_key, expected_payload):
+        global routing_key, amqp_body
+        self.assertEquals(
+            routing_key, expected_key, "Routing key is not expected! expect {0}, get{1}"
+            .format(expected_key, routing_key))
+        try:
+            amqp_body_json = fit_common.json.loads(amqp_body)
+        except:
+            logs.error("FAILURE - The message body is not json format!")
+            return False
+        try:
+            self.assertEquals(
+                amqp_body_json['version'], expected_payload['version'],
+                "version field not correct! expect {0}, get {1}"
+                .format(expected_payload['version'], amqp_body_json['version']))
+            self.assertEquals(
+                amqp_body_json['typeId'], expected_payload['typeId'],
+                "typeId field not correct!  expect {0}, get {1}"
+                .format(expected_payload['typeId'], amqp_body_json['typeId']))
+            self.assertEquals(
+                amqp_body_json['action'], expected_payload['action'],
+                "action field not correct!  expect {0}, get {1}"
+                .format(expected_payload['action'], amqp_body_json['action']))
+            self.assertEquals(
+                amqp_body_json['severity'], expected_payload['severity'],
+                "serverity field not correct!"
+                .format(expected_payload['severity'], amqp_body_json['severity']))
+            self.assertNotEquals(amqp_body_json['createdAt'], {}, "createdAt field is empty!")
+            self.assertNotEquals(amqp_body_json['data'], {}, "data field is empty!")
+        except ValueError:
+            logs.error("FAILURE - expected key is missing in the AMQP message!")
+            return False
+        return True
+
+    def amqp_callback(self, ch, method, properties, body):
+        logs.debug_3("Routing Key %r:" % method.routing_key)
+        logs.debug_3(body.__str__())
+        global amqp_message_received
+        global routing_key, amqp_body
+        amqp_message_received = True
+        amqp_body = body
+        routing_key = method.routing_key
+
+    def test_api_create_and_check_racks(self):

```

Comment:
@hohene Yes, I had another PR https://github.com/RackHD/RackHD/pull/460 for API and AMQP test. But the FIT was changed so much that since I open the original one. I have to close this PR  and submit another based on the new FIT file architecture very soon.